Warnings of "IllegalStateException: unexpected docvalues type NONE for field '_parent'" on upgraded index

**Elasticsearch version**: 5.5.0

**Plugins installed**: [ x-pack ]

**JVM version** (`java -version`): 1.8.0_92

**OS version** (`uname -a` if on a Unix-like system): Windows 10

**Description of the problem including expected versus actual behavior**:
_parent fields on upgraded indices seem to cause, IMHO undesirable and unexpected, IllegalStateException warnings in the logs on startup even though the migration checker and documentation suggests these fields require no action.

If we ignore the warning, is it a concern that the upgraded _parent field is not a doc value? For performance perhaps?

**Steps to reproduce**:
 1. Create an index on 2.4.1
````
PUT test-index
{
  "mappings": {
    "my-child": {
      "properties": {
        "child-1": {
          "type": "long"
        }
      },
      "_parent": {
        "type": "my-parent"
      }
    },
    "my-parent": {}
  }
}
````
 2. Add parent and child documents
````
PUT test-index/my-parent/1
{
  "field-1": "testing"
}

PUT test-index/my-child/1?parent=1
{
  "child-1": 20
}
````
 3. Upgrade to 5.5.0 (or copy 2.4.1 data folder to 5.5.0 data folder) and start

**Provide logs (if relevant)**:
Logs showing the index getting upgraded without incident:
````
[2017-07-24T11:21:29,740][INFO ][o.e.c.u.IndexFolderUpgrader] [test-index/6_nfai1tT0C9oZpnj6ztyw] upgrading [D:\work\Elasticsearch\elasticsearch-5.5.0 - Upgraded\data\batman\nodes\0\indices\test-index] to new naming convention
[2017-07-24T11:21:29,742][INFO ][o.e.c.u.IndexFolderUpgrader] [test-index/6_nfai1tT0C9oZpnj6ztyw] moved from [D:\work\Elasticsearch\elasticsearch-5.5.0 - Upgraded\data\batman\nodes\0\indices\test-index] to [D:\work\Elasticsearch\elasticsearch-5.5.0 - Upgraded\data\batman\nodes\0\indices\6_nfai1tT0C9oZpnj6ztyw]
````
The exception that comes shortly thereafter:
````
[2017-07-24T11:21:34,221][WARN ][o.e.i.w.ShardIndexWarmerService] [Z6tXLYo] [test-index][3] failed to warm-up global ordinals for [_parent]
org.elasticsearch.ElasticsearchException: java.util.concurrent.ExecutionException: java.lang.IllegalStateException: unexpected docvalues type NONE for field '_parent' (expected one of [SORTED, SORTED_SET]). Re-index with correct docvalues type.
        at org.elasticsearch.index.fielddata.plain.SortedSetDVOrdinalsIndexFieldData.loadGlobal(SortedSetDVOrdinalsIndexFieldData.java:120) ~[elasticsearch-5.5.0.jar:5.5.0]
        at org.elasticsearch.index.fielddata.plain.SortedSetDVOrdinalsIndexFieldData.loadGlobal(SortedSetDVOrdinalsIndexFieldData.java:45) ~[elasticsearch-5.5.0.jar:5.5.0]
        at org.elasticsearch.index.IndexWarmer$FieldDataWarmer.lambda$warmReader$1(IndexWarmer.java:141) ~[elasticsearch-5.5.0.jar:5.5.0]
        at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:569) [elasticsearch-5.5.0.jar:5.5.0]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_92]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_92]
        at java.lang.Thread.run(Thread.java:745) [?:1.8.0_92]
Caused by: java.util.concurrent.ExecutionException: java.lang.IllegalStateException: unexpected docvalues type NONE for field '_parent' (expected one of [SORTED, SORTED_SET]). Re-index with correct docvalues type.
        at org.elasticsearch.common.cache.Cache.computeIfAbsent(Cache.java:404) ~[elasticsearch-5.5.0.jar:5.5.0]
        at org.elasticsearch.indices.fielddata.cache.IndicesFieldDataCache$IndexFieldCache.load(IndicesFieldDataCache.java:154) ~[elasticsearch-5.5.0.jar:5.5.0]
        at org.elasticsearch.index.fielddata.plain.SortedSetDVOrdinalsIndexFieldData.loadGlobal(SortedSetDVOrdinalsIndexFieldData.java:115) ~[elasticsearch-5.5.0.jar:5.5.0]
        ... 6 more
Caused by: java.lang.IllegalStateException: unexpected docvalues type NONE for field '_parent' (expected one of [SORTED, SORTED_SET]). Re-index with correct docvalues type.
        at org.apache.lucene.index.DocValues.checkField(DocValues.java:212) ~[lucene-core-6.6.0.jar:6.6.0 5c7a7b65d2aa7ce5ec96458315c661a18b320241 - ishan - 2017-05-30 07:29:46]
        at org.apache.lucene.index.DocValues.getSortedSet(DocValues.java:306) ~[lucene-core-6.6.0.jar:6.6.0 5c7a7b65d2aa7ce5ec96458315c661a18b320241 - ishan - 2017-05-30 07:29:46]
        at org.elasticsearch.index.fielddata.plain.SortedSetDVBytesAtomicFieldData.getOrdinalsValues(SortedSetDVBytesAtomicFieldData.java:53) ~[elasticsearch-5.5.0.jar:5.5.0]
        at org.elasticsearch.index.fielddata.ordinals.GlobalOrdinalsBuilder.build(GlobalOrdinalsBuilder.java:63) ~[elasticsearch-5.5.0.jar:5.5.0]
        at org.elasticsearch.index.fielddata.plain.SortedSetDVOrdinalsIndexFieldData.localGlobalDirect(SortedSetDVOrdinalsIndexFieldData.java:127) ~[elasticsearch-5.5.0.jar:5.5.0]
        at org.elasticsearch.index.fielddata.plain.SortedSetDVOrdinalsIndexFieldData.localGlobalDirect(SortedSetDVOrdinalsIndexFieldData.java:45) ~[elasticsearch-5.5.0.jar:5.5.0]
        at org.elasticsearch.indices.fielddata.cache.IndicesFieldDataCache$IndexFieldCache.lambda$load$1(IndicesFieldDataCache.java:157) ~[elasticsearch-5.5.0.jar:5.5.0]
        at org.elasticsearch.common.cache.Cache.computeIfAbsent(Cache.java:401) ~[elasticsearch-5.5.0.jar:5.5.0]
        at org.elasticsearch.indices.fielddata.cache.IndicesFieldDataCache$IndexFieldCache.load(IndicesFieldDataCache.java:154) ~[elasticsearch-5.5.0.jar:5.5.0]
        at org.elasticsearch.index.fielddata.plain.SortedSetDVOrdinalsIndexFieldData.loadGlobal(SortedSetDVOrdinalsIndexFieldData.java:115) ~[elasticsearch-5.5.0.jar:5.5.0]
        ... 6 more
````

--- a/core/src/main/java/org/elasticsearch/index/mapper/ParentFieldMapper.java
+++ b/core/src/main/java/org/elasticsearch/index/mapper/ParentFieldMapper.java
@@ -65,7 +65,7 @@ public class ParentFieldMapper extends MetadataFieldMapper {
             FIELD_TYPE.setIndexOptions(IndexOptions.NONE);
             FIELD_TYPE.setHasDocValues(true);
             FIELD_TYPE.setDocValuesType(DocValuesType.SORTED);
-            FIELD_TYPE.setEagerGlobalOrdinals(true);
+            FIELD_TYPE.setEagerGlobalOrdinals(false);
             FIELD_TYPE.freeze();
         }
     }

--- a/core/src/main/java/org/elasticsearch/index/mapper/ParentFieldMapper.java
+++ b/core/src/main/java/org/elasticsearch/index/mapper/ParentFieldMapper.java
@@ -78,6 +78,8 @@ public class ParentFieldMapper extends MetadataFieldMapper {
 
         public Builder(String documentType) {
             super(Defaults.NAME, new ParentFieldType(Defaults.FIELD_TYPE, documentType), Defaults.FIELD_TYPE);
+            // Defaults to true
+            eagerGlobalOrdinals(true);
             this.documentType = documentType;
             builder = this;
         }

--- a/core/src/test/java/org/elasticsearch/index/mapper/ParentFieldMapperTests.java
+++ b/core/src/test/java/org/elasticsearch/index/mapper/ParentFieldMapperTests.java
@@ -124,6 +124,8 @@ public class ParentFieldMapperTests extends ESSingleNodeTestCase {
         Set<String> allFields = new HashSet<>(mapperService.simpleMatchToIndexNames("*"));
         assertTrue(allFields.contains("_parent"));
         assertFalse(allFields.contains("_parent#null"));
+        MappedFieldType fieldType = mapperService.fullName("_parent");
+        assertFalse(fieldType.eagerGlobalOrdinals());
     }
 
     private static int getNumberOfFieldWithParentPrefix(ParseContext.Document doc) {

